File: frontend/src/schemas/annotation.schema.ts
import { z } from 'zod'

export const AnnotationSchema = z.object({
  id: z.string().uuid(),
  content: z.string(),
  date: z.date(),
  type: z.enum(['note', 'catalyst']),
  tradeId: z.string().uuid(),
})

export const AnnotationCreateSchema = AnnotationSchema.omit({ id: true, date: true })

File: backend/domain/annotation/annotation_schema.py
from datetime import datetime
from typing import Optional
from core.base_schema import BaseSchema
from typing import Literal


class AnnotationBase(BaseSchema):
    content: str
    annotation_type: Literal["catalyst", "note", "managementNote"]


class AnnotationCreate(AnnotationBase):
    trade_id: str


class AnnotationRead(AnnotationCreate):
    id: str
    date: datetime


class AnnotationUpdate(AnnotationBase):
    annotation_type: Optional[Literal["catalyst", "note", "managementNote"]] = None
    content: Optional[str] = None
    date: Optional[datetime] = None
